consumer/consumer_test.go

```diff
@@ -62,14 +66,15 @@ func (t *TestCase) TestSetup() []error {
 		errs = append(errs, errList...)
 	}
 
-	consumer := mocks.MockConsumer{}
-	t.inv, err = New(t.completedConfig, nil, t.logger)
+	// Create mock consumer first
+	mockConsumer := &mocks.MockConsumer{}
+
+	// Use NewWithConsumer to avoid creating a real Kafka connection
+	t.inv, err = NewWithConsumer(t.completedConfig, nil, t.logger, mockConsumer)
```

Comment:
so much this...been meaning to look at this so the tests dont dump 100's of failed kafka connection attempts buts still pass haha. Thanks!